tests/template_tests/filter_tests/test_floatformat.py

```diff
@@ -181,12 +182,21 @@ def test_too_many_digits_to_render(self):
             "-1E10000000000000000",
             "1e10000000000000000",
             "-1e10000000000000000",
-            "1" + "0" * 1_000_000,
         ]
         for value in cases:
             with self.subTest(value=value):
                 self.assertEqual(floatformat(value), value)
 
+    def test_too_many_digits_to_render_very_long(self):
+        value = "1" + "0" * 1_000_000
+        if PYPY:
+            # PyPy casts decimal parts to int, which reaches the integer string
+            # conversion length limit (default 4300 digits, CVE-2020-10735).
+            with self.assertRaises(ValueError):
```

Comment:
I think it's worth keeping this assertion to ensure that we're not vulnerable to CVE-2024-41989 on PyPy.